code/game/objects/effects/decals/cleanable/misc.dm

```diff
@@ -82,7 +82,16 @@
 	random_icon_states = list("vomit_1", "vomit_2", "vomit_3", "vomit_4")
 	var/list/viruses = list()
 
-/obj/effect/decal/cleanable/vomit/attack_hand(var/mob/user)
+/obj/effect/decal/cleanable/vomit/Initialize(mapload, list/virus_list, virus_chance = 20)
+	. = ..()
+	if(virus_list && virus_list.len)
+		for(var/datum/disease/D in virus_list)

```

Comment:
I'll do it, but I'm still not convinced it actually saves worthwhile CPU cycles, in most cases.